packages/core/src/oidc/utils.ts

```diff
@@ -27,19 +27,21 @@ const isOrigin = (value: string) => {
   }
 };
 
+export const validateCorsAllowedOrigins = (value: unknown) => {
+  if (Array.isArray(value) && value.some((origin) => !isOrigin(origin))) {
+    throw new errors.InvalidClientMetadata(CustomClientMetadataKey.CorsAllowedOrigins);
+  }
+};
+
 export const validateCustomClientMetadata = (key: string, value: unknown) => {
   const result = customClientMetadataGuard.pick({ [key]: true }).safeParse({ [key]: value });
 
   if (!result.success) {
     throw new errors.InvalidClientMetadata(key);
   }
 
-  if (
-    key === CustomClientMetadataKey.CorsAllowedOrigins &&
-    Array.isArray(value) &&
-    value.some((origin) => !isOrigin(origin))
-  ) {
-    throw new errors.InvalidClientMetadata(CustomClientMetadataKey.CorsAllowedOrigins);
+  if (key === CustomClientMetadataKey.CorsAllowedOrigins) {
+    validateCorsAllowedOrigins(value);
```

Comment:
The only reference of a util method is another util method. This implementation seems a little bit odd.

Reply:
3 references.

another 2:

<img width="568" alt="image" src="https://user-images.githubusercontent.com/10594507/163097055-ef9cd890-bd1a-4b23-83d1-b339b5c4e676.png">
<img width="502" alt="image" src="https://user-images.githubusercontent.com/10594507/163097057-09bb7184-15ea-47ab-96ac-0be4765b7be5.png">

Reply:
Refactored. Implement it in another way. PTAL again.